text/XXXX-supergroups.md

```diff
@@ -0,0 +1,275 @@
+functions.  Many inline functions do not carry the same amount of information as a non inlined one.
+For instance take this function:
+
+```
+template <class T> const T &min(const T &a, const T &b)
+{
+    return !(b < a) ? a : b;
+}
+```
+
+When invoked with an `int` the function when not inlined is typically recoded as
+`module::min<int>(const int &, const int &)` or `min(const int &, const int &)`
+in the stack trace.  However when inlined the function on Windows platforms
+might only be called `min`.  However that's not true in the general case either.
+Many functions do retain their types, some do not.
+
+This problem becomes even larger when dealing with compiler generated code.  For instance C++
+knows a lot of different destructors and they can all look very different and they will look
+different depending on if they are inlined or not.
+
+As such from one release to another the fingerprint generated for a stack might change entirely.
+
+### Platform Unstable Stacks
+
+Related to the former case another complication arises from platform provided primitives that might
+end up in the stack trace.  Imagine for instance an application written in Rust that runs on macOS,
+Windows and Linux and has a trivial user introduced panic in a function invoked from an event loop.
+The event loop part of the stack will look different on every platform, yet the user code on top
+will look the very same.  On Linux it might use some epoll abstraction, in Windows IOCP or kqueue on
+mac.  Similar problems happy with mutexes, thread spawning code or more.
+
+None of the platform differences are relevant to the user bug, yet the same issue will create three
+different fingerprints, one per platform.
+
+We can also not universally exclude platform specific code because there might very well be a bug
+in this threading code.  For instance you would not want to have all the event handling code be
+excluded from stack traces for the case that the bug is in fact in the platform specific event
+handling code.
+
+## Proposed Changes
+
+The relationship of event to issue is very strong in Sentry.  There are directly workflows associated
+with it and there are quite significant technical hurdles to this.  The proposal is to not touch this
+association but instead augment it on a higher level.  The idea is to actually create these Sentry
+issues still based on the unique fingerprint but to lower the consequences of such a group being
+created.  Related fingerprints could then at regular intervals be sweeped up in to larger super
+groups.
+
+This means that from a user's point of view the individual groups still exist, but the main way to
+interact with them would be larger groups that group around the already existing smaller groups.
+Since the events stay associated with their fingerprint created group the event itself is immutable
+and stays that way, but the merge just becomes an association of that group with the supergroup.
```

Comment:
How do you see supergroups being represented in the datamodel? Are they an issue made of other 'similiar' issues?

Reply:
@markstory i think supergroups might be just a special case of a re-built merge where merges are multiple hashes being associated to a group by some mutable mechanism. The difference between a merge and a supergroup in that sense is, that a supergroup carries additional metadata that gives the merge some meaning.

There is some related thinking happening here: https://github.com/getsentry/rfcs/pull/30